java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java

```diff
@@ -653,6 +653,7 @@ private K disjunctResults(List<K> results) {
                 boolean alreadyLogged = false;
                 try {
                     v++;
+                    term.termContext().setTopConstraint(null); //To remove leftover constraint from previous step
```

Comment:
Down on line 718/768, we're doing `nextQueue.add(result)`, where `result` is a `ConstrainedTerm`. So `result` could already have a constraint which we should be keeping, will this throw that constraint away? Or is that constraint different than the one in `_.termContext()`?

Reply:
It's different. ConstrainedTerm is immutable. While TermContaxt is just a bag of mutable fields. All of them should be evaluated, and (ideally) either replaced with parameters or made immutable.